sdk/provisioning/Azure.Provisioning/src/Generated/ResourceGroup.cs

```diff
@@ -0,0 +1,118 @@
+    private readonly BicepValue<string> _managedBy;
+
+    /// <summary>
+    /// Resource tags.
+    /// </summary>
+    public BicepDictionary<string> Tags { get => _tags; set => _tags.Assign(value); }
+    private readonly BicepDictionary<string> _tags;
+
+    /// <summary>
+    /// Fully qualified resource ID for the resource. Ex -
+    /// /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}.
+    /// </summary>
+    public BicepValue<ResourceIdentifier> Id { get => _id; }
+    private readonly BicepValue<ResourceIdentifier> _id;
+
+    /// <summary>
+    /// The provisioning state.
+    /// </summary>
+    public BicepValue<string> ResourceGroupProvisioningState { get => _resourceGroupProvisioningState; }
+    private readonly BicepValue<string> _resourceGroupProvisioningState;
+
+    /// <summary>
+    /// Azure Resource Manager metadata containing createdBy and modifiedBy
+    /// information.
+    /// </summary>
+    public BicepValue<SystemData> SystemData { get => _systemData; }
+    private readonly BicepValue<SystemData> _systemData;
+
+    /// <summary>
+    /// Creates a new ResourceGroup.
+    /// </summary>
+    /// <param name="resourceName">Name of the ResourceGroup.</param>
+    /// <param name="resourceVersion">Version of the ResourceGroup.</param>
+    /// <param name="context">Provisioning context for this resource.</param>
+    public ResourceGroup(string resourceName, string? resourceVersion = default, ProvisioningContext? context = default)
+        : base(resourceName, "Microsoft.Resources/resourceGroups", resourceVersion, context)
+    {
+        _name = BicepValue<string>.DefineProperty(this, "Name", ["name"], isRequired: true);
+        _location = BicepValue<AzureLocation>.DefineProperty(this, "Location", ["location"], isRequired: true);
+        _managedBy = BicepValue<string>.DefineProperty(this, "ManagedBy", ["managedBy"]);
+        _tags = BicepDictionary<string>.DefineProperty(this, "Tags", ["tags"]);
+        _id = BicepValue<ResourceIdentifier>.DefineProperty(this, "Id", ["id"], isOutput: true);
+        _resourceGroupProvisioningState = BicepValue<string>.DefineProperty(this, "ResourceGroupProvisioningState", ["properties", "provisioningState"], isOutput: true);
+        _systemData = BicepValue<SystemData>.DefineProperty(this, "SystemData", ["systemData"], isOutput: true);
+    }
+
+    /// <summary>
+    /// Creates a reference to an existing ResourceGroup.
+    /// </summary>
+    /// <param name="resourceName">Name of the ResourceGroup.</param>
+    /// <param name="resourceVersion">Version of the ResourceGroup.</param>
+    /// <returns>The existing ResourceGroup resource.</returns>
+    public static ResourceGroup FromExisting(string resourceName, string? resourceVersion = default) =>
+        new(resourceName, resourceVersion) { IsExistingResource = true };
```

Comment:
There are two other "magic" values worth calling out.  Anything deriving from `Resource` can be `existing` so that effectively turns everything but the `Name` into an output reference.